agent/engine/docker_container_engine.go

```diff
@@ -146,10 +147,13 @@ func NewDockerGoClient(clientFactory dockerclient.Factory, authType string, auth
 		return nil, err
 	}
 
+	cfg := config.EnvironmentConfig()
```

Comment:
Rather than calling `config.EnvironmentConfig()` directly (which won't handle merging things from `FileConfig` or `DefaultConfig` properly), it'd be better to pass in the existing `config.Config` and use that.  In `NewDockerGoClient`, you can change the `authData` argument (which is a subset of the config) to just take in the whole `config.Config`.